diff --git a/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/MethodDeclaration.java b/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/MethodDeclaration.java
index b00d0de..93ce145 100644
--- a/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/MethodDeclaration.java
+++ b/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/MethodDeclaration.java
@@ -834,7 +834,11 @@ public class MethodDeclaration extends BodyDeclaration {
 	 * @deprecated In the JLS8 API, this method is replaced by {@link #thrownExceptionTypes()}.
 	 */
 	public List thrownExceptions() {
-		return internalThrownExceptions();
+		if (getAST().apiLevel() >= AST.JLS8) {
+			return thrownExceptionTypes();
+		} else {
+			return internalThrownExceptions();
+		}
 	}
 
 	/**
